compiler/cpp/test/lexer/LexerTest.cpp

```diff
@@ -0,0 +1,435 @@
+//
+// Created by Marc on 30.05.2021.
+//
+
+#include <vector>
+#include <lexer/Lexer.h>
+#include <gtest/gtest.h>
+#include "../util/LanguageDescriptor.h"
+#include "../util/FileReader.h"
+
+struct LexerParams {
+    const std::string fileName;
+    std::vector<TokenType> expectedTokensLineComment;
+    std::vector<TokenType> expectedTokensBlockComment;
+};
+
+const std::vector<LanguageDescriptor> LANGUAGES {
+    LanguageDescriptor("yml", "#", "", ""),
+    LanguageDescriptor("cpp", "//", "/*", "*/"),
+    LanguageDescriptor("html", "", "<!--", "-->"),
+    LanguageDescriptor("py", "#", "", "")
+};
+
+const LexerParams LEXER_TEST_PARAMETERS[] = {
+    {
+            "brace-close",
+            {
+                TokenType::TOK_COM_LINE_IDEN,
```

Comment:
The first value of every object here is the exactly the same: `TOM_COM_LINE_IDEN` and `TOM_COM_BLOCK_IDEN_OPEN`. Can you somehow extract this?

Reply:
I could extract it, but I wanted it to be extensible for the future. Is there anything against it?

Reply:
No, nothing against it. I guess I've misunderstood your `LEXER_TEST_PARAMETERS[]`. Smart idea to do it as a parameterized test 👍